### PR TITLE
11272

### PR DESCRIPTION
**Pull request by faustino-garcia**
Add ability to delete ingredients

> Acceptance Criteria:

- [x] Add "Delete Ingredient" button next to the "Update Ingredient" (red) button

- [x] After deleted, return to the ingredient listing with the flash message "Ingredient was deleted successfully"

- [x] Ingredient must be deleted from all recipes

### CODE REVIEW

**Review by vicmaburrito — CHANGES_REQUESTED** [review]
## Please change development instead initial branch to merge your feature branch

**Comment by coderdave** [comment]
is this up to date with development?

**Comment by ArmandoD97** [comment]
> is this up to date with development?

Fixed now

**Review by vicmaburrito — APPROVED** [review]

**Review by AlexRS90 — APPROVED** [review]
## Hello team

Good job working in the requested features!
Everything is working as expected and all the test are also passing, there is nothing else to say other than... It's time to merge!!! 🚀 🎸 🚀 🎸

File: app/controllers/ingredients_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class IngredientsController < ApplicationController
-  before_action :set_ingredient, only: [:edit, :update, :show]
+  before_action :set_ingredient, only: [:edit, :update, :show, :destroy] #Added destroy here
```
Comment (coderdave) [inline app/controllers/ingredients_controller.rb:2]: you don't need comment "added destroy here"

File: app/views/ingredients/_form.html.erb [hunk 2]
```diff
@@ -8,12 +8,17 @@
...
         </div>
         <div class="col-md-8">
-          <%= f.text_field :name, class: "form-control", placeholder: "Enter ingredient", autofocus: true, class: "form-control" %>
+          <%= f.text_field :name, placeholder: "Enter ingredient", autofocus: true, class: "form-control" %>
         </div>
       </div>
       <div class="col mb-3">
         <%= f.submit(@ingredient.new_record? ? "Add Ingredient" : "Update Ingredient",
                                       class: "btn btn-primary btn-lg") %>
+        <% if  logged_in? && (current_chef.admin?) && (@ingredient.new_record? ==false) %>
```
Comment (coderdave) [inline app/views/ingredients/_form.html.erb:17]: this is minor, but you don't need parentheses around the `current_chef.admin?` check and the `(@ingredient.new_record?` check

File: test/integration/ingredients_delete_test.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    assert_redirected_to ingredients_path
+    assert_not flash.empty?
+  end
+
+  test "Succesfully delete the association between recipe and ingredient after deleting a recipe" do
+    sign_in_as(@admin, "password")
+    delete recipe_path(@recipe)
+    assert RecipeIngredient.where(recipe_id: @recipe).count == 0
+  end
+
```
Comment (coderdave) [inline test/integration/ingredients_delete_test.rb:30]: add another test to make sure that non-admin users can't delete ingredients

**Review by coderdave — CHANGES_REQUESTED** [review]
Good job! Please make those final changes so I can merge to master and send to testing.

File: test/integration/ingredients_delete_test.rb (hunk 3 above)
Comment (faustino-garcia) [reply test/integration/ingredients_delete_test.rb:30]: Thanks for the Feedback, I follow your advice and added a new test

File: app/controllers/ingredients_controller.rb (hunk 1 above)
Comment (faustino-garcia) [reply app/controllers/ingredients_controller.rb]: The comment is gone in the new commit

**Review by coderdave — APPROVED** [review]

**Pull request merged**